vector-im/element-x-android
# [Room Details] Implement room details screen

## What

- Implements a new Room Details screen containing the info of for the selected room.
- Updates Preferences in design system to match the designs.
- Add 'share room' action to the screen.

## Why 

Implements #251 .

## Test

1. Go to any room.
2. Tap on the room name in the appbar.
3. You should see the updated details for that room.

Note that the actions of the new screen don't do anything at the moment.
